source/resolve-conflicts.ts

```diff
@@ -0,0 +1,129 @@
+		const lines = document.querySelectorAll('.CodeMirror .conflict-gutter-marker');
+		let stepBranch = '';
+		for (const line of lines) {
+			const lineNumber = Number(line.parentElement.parentElement.textContent) - 1;
+			let cl = `rgh-conflict c-${this.conflictNumber} `;
+			if (line.classList.contains('js-start')) {
+				cl += 'rgh-separator rgh-start rgh-incoming-branch rgh-current-branch';
+				stepBranch = 'rgh-incoming-branch';
+				this.addWidgetToLine(this.conflictNumber, lineNumber);
+			} else if (line.classList.contains('js-middle')) {
+				cl += 'rgh-separator rgh-middle rgh-incoming-branch rgh-current-branch';
+				stepBranch = 'rgh-current-branch';
+			} else if (line.classList.contains('js-end')) {
+				cl += 'rgh-separator rgh-end rgh-incoming-branch rgh-current-branch';
+				this.conflictNumber += 1;
+			} else if (line.classList.contains('js-line')) {
+				cl += 'rgh-middle ' + stepBranch;
+			}
+
+			this.editor.addLineClass(lineNumber, '', cl);
+		}
+	}
+
+	// Create and add widget to a line
+	addWidgetToLine(conflictNumber: number, lineNumber: number) {
+		const widget = this.newWidget(conflictNumber);
+		this.editor.addLineWidget(lineNumber, widget, {above: true});
+	}
+
+	// Create and return conflict resolve widget for specific conflict
+	newWidget(conflictNumber: number) {
+		const widget = document.createElement('div');
+		widget.classList.add('rgh-resolver');
+		widget.style.fontWeight = 'bold';
+
+		const link = (branch, conflictNumber) => {
+			const link = document.createElement('a');
+			link.href = `#accept${branch}`;
+			link.textContent = `Accept ${branch} Change`;
+			link.addEventListener('click', e => {
+				e.preventDefault();
+				this.acceptBranch(branch, conflictNumber);
+			});
+			return link;
+		};
+
+		widget.append(
+			link('Current', conflictNumber),
+			' | ',
+			link('Incoming', conflictNumber),
+			' | ',
+			link('Both', conflictNumber)
+		);
+		return widget;
+	}
+
+	// Accept one or both of branches and remove unnecessary lines
+	acceptBranch(branch: string, conflictNumber: number) {
+		if (branch === 'Both') {
+			this.removeLines('.rgh-separator', conflictNumber);
+		} else if (branch === 'Incoming') {
+			this.removeLines('.rgh-current-branch', conflictNumber);
+		} else if (branch === 'Current') {
+			this.removeLines('.rgh-incoming-branch', conflictNumber);
+		}
+	}
+
+	// Remove Lines of a conflict that matchs given selector
+	removeLines(selector, conflictNumber) {
+		const lines = [...document.querySelectorAll(`.rgh-conflict.c-${conflictNumber}${selector}`)];
+
+		// Remove lines in revese order
+		lines.reverse();
+
+		for (const line of lines) {
+			const element = line.querySelector('.CodeMirror-linenumber');
+			const lineNumber = Number(element.textContent);
+			this.removeline(lineNumber - 1);
+		}
+	}
+
+	// Remove a line
+	removeline(lineNumber) {
+		this.replaceLine('', lineNumber);
+	}
+
+	// Replace line with given text
+	replaceLine(newLine, lineNumber) {
```

Comment:
It looks like this method can be merged into `removeline`

Reply:
I separate these two functions for a reason, I'll want to replace start and end line of conflict with new line for better understanding, for example:
```
<<<<<<< branch6
Lorem Ipsum is simply 6
=======
Lorem Ipsum is simply 5
>>>>>>> branch5
```
to 
```
<<<<<<< branch6 - Incoming Change
Lorem Ipsum is simply 6
=======
Lorem Ipsum is simply 5
>>>>>>> branch5 - Current Change
```